marlowe/src/Language/Marlowe/Core/V1/Semantics/Types.hs

```diff
@@ -307,6 +314,13 @@ data State = State { accounts    :: Accounts
 newtype Environment = Environment { timeInterval :: TimeInterval }
   deriving stock (Haskell.Show,Haskell.Eq,Haskell.Ord)
 
+environment :: UTCTime -> UTCTime -> Environment
+environment start end = Environment (utcTimeToPOSIXTime start, utcTimeToPOSIXTime end)
+
+isTimedOut :: POSIXTime -> Environment -> Bool
+isTimedOut t Environment {timeInterval = (_,b)} | t  > b = True
+isTimedOut _ _ = False
```

Comment:
```suggestion
```
This doesn't seem to be used, and also I think it is reversed, if it is the timeout, then it should be before the time interval right?